lib/fast_jsonapi/serialization_core.rb

```diff
@@ -68,23 +67,38 @@ def meta_hash(record, params = {})
       def record_hash(record, fieldset, includes_list, params = {})
         if cache_store_instance
           cache_opts = record_cache_options(cache_store_options, fieldset, includes_list, params)
-          record_hash = cache_store_instance.fetch(record, **cache_opts) do
-            temp_hash = id_hash(id_from_record(record, params), record_type, true)
-            temp_hash[:attributes] = attributes_hash(record, fieldset, params) if attributes_to_serialize.present?
-            temp_hash[:relationships] = relationships_hash(record, cachable_relationships_to_serialize, fieldset, includes_list, params) if cachable_relationships_to_serialize.present?
-            temp_hash[:links] = links_hash(record, params) if data_links.present?
-            temp_hash
+          klass = self.name
+          param = [record, cache_opts, klass]
+          ::BatchLoader.for(param).batch do |batch_params, loader|
+            cache_results = cache_store_instance.fetch(batch_params)
```

Comment:
what do we assume to be the class of `cache_store_instance`?

Reply:
why do we need to pass all three of these `[record, cache_opts, klass]` to `fetch`?

Reply:
It's a Redis cache store instance, specified in serializer#cache_options, could be `Rails.cache`